app/src/main/java/com/example/android/architecture/blueprints/todoapp/di/DataModules.kt

```diff
@@ -44,8 +47,15 @@ object RepositoryModule {
     fun provideTaskRepository(
         @NetworkTaskDataSource remoteDataSource: NetworkDataSource,
         database: ToDoDatabase,
+        @DefaultDispatcher dispatcher: CoroutineDispatcher,
```

Comment:
Nit: Update with the implementation that provides the `TaskDao` instead of the repo

Reply:
I've replaced `provides` with `binds` for the `TaskRepository` which is quite a bit simpler.